exonum/src/node/consensus.rs

```diff
@@ -568,8 +563,17 @@ impl NodeHandler {
                 return;
             }
         });
+        let mut fork = self.blockchain.fork();
+        {
+            let mut schema = Schema::new(&mut fork);
+            schema.transactions_pool_mut().insert(hash);
+            schema.transactions_mut().put(&hash, msg);
+        }
+        self.blockchain.merge(fork.into_patch()).expect(
+            "Unable to save transaction to persistent pool.",
+        );
```

Comment:
As far as I can see, this code is duplicated (thrice :upside_down_face:). It is worth to create a separate function.

Reply:
Body of `handle_tx` function duplicate `handle_tx_incomming` body. So we should fully refactor this methods (and this will be done in the multithread tx validation). The third duplication is a bit different (it add many tx at once)